igvm/src/c_api.rs

```diff
@@ -0,0 +1,763 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+//
+// Copyright (c) 2023 SUSE LLC
+//
+// Author: Roy Hopkins <rhopkins@suse.de>
+#![allow(unsafe_code)]
+
+use std::{
+    collections::BTreeMap,
+    ptr::null,
+    sync::{
+        atomic::{AtomicI64, Ordering},
+        Mutex,
+    },
+};
+
+use crate::{Error, IgvmFile};
+use open_enum::open_enum;
+
+#[open_enum]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(i64)]
+pub enum IgvmApiError {
```

Comment:
I think this is safe to return directly. Since it's `repr(transparent)` as an i64 with open_enum, there's no need for the additional IgvmResult type. That should help make the code cleaner and reduce the need for .0.

See https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent and https://docs.rs/open-enum/0.4.0/open_enum/#integer-type.

Reply:
After some thought I've now replaced the `IgvmApiError` enum with constants, resulting in `#define`'s in the generated C code. The functions now return either an `i32` or a handle type which is actually an `i32`. This makes the C return codes more obvious and still benefits from removing the additional type and the need for `.0`.